runtime/runtime/src/actions.rs

```diff
@@ -183,6 +183,7 @@ pub(crate) fn action_function_call(
         // TODO
         if let Some(outcome) = outcome {
             result.gas_burnt += outcome.burnt_gas;
+            result.gas_used += outcome.used_gas;
```

Comment:
```suggestion
            result.gas_used += outcome. burnt_gas;
```

Reply:
Can you explain why we should use `burnt_gas` instead of `used_gas` in case of failure? I think VM maintains invariance that  `burnt_gas <= used_gas` all the time.

Reply:
In case of failure attached gas (which is what makes the difference between `used_gas` and `burnt_gas`) is undone, as if it did not happen. So in case of failure `gas_used` and `gas_burnt` of `ActionResult` should be incremented by the same amount, because not gas "attachment" did not happen (it happened but was reverted).

Reply:
The proposed edit does not break the `burnt_gas <= used_gas` invariance .

Reply:
sounds like it's safe to `assert_eq!(outcome.used_gas, outcome.burnt_gas);`

Reply:
Runtime in `generate_refund_receipts` takes care of using proper value for refunds. It uses `gas_used` for success and `gas_burnt` for failures. So it's not an issue to return a real `gas_used` instead of the `gas_burnt` into ActionResult for FunctionCalls.

Reply:
Please add this explanation as a comment on that line.